uploader/pantheon.py

```diff
@@ -206,102 +201,114 @@ def process_file(path, filetype):
+
+    if parent_dir_str:
+        url += '/' + parent_dir_str
+
+    logger.debug('base name: %s', base_name)
+
+    # Asciidoc content (treat as a module)
+    if isModule:
+        url += '/' + path.name
+        logger.debug('url: %s', url)
+        jcr_primary_type = "pant:module"
+        data = _generate_data(jcr_primary_type, base_name, path.name, asccidoc_type="nt:file")
+        # This is needed to add a new module version, otherwise it won't be handled
+        data[":operation"] = "pant:newModuleVersion"
+        files = {'asciidoc': ('asciidoc', open(path, 'rb'), 'text/x-asciidoc')}
+
+        # Minor question: which is correct, text/asciidoc or text/x-asciidoc?
+        # It is text/x-asciidoc. Here's why:
+        # https://tools.ietf.org/html/rfc2045#section-6.3
+        # Paraphrased: "If it's not an IANA standard, use the 'x-' prefix.
+        # Here's the list of standards; text/asciidoc isn't in it.
+        # https://www.iana.org/assignments/media-types/media-types.xhtml#text
+
+        if not args.dry:
+            r = requests.post(url, headers=HEADERS, data=data, files=files, auth=(args.user, pw))
+            _print_response('module', path, r.status_code, r.reason)
+        processed_files.append(path)
+        logger.debug('')
+    elif isResource:
+        if os.path.islink(path):
+            target = str(os.readlink(path))
             url += '/' + path.name
             logger.debug('url: %s', url)
-            jcr_primary_type = "pant:module" if isModule else "pant:title"
-            data = _generate_data(jcr_primary_type, base_name, path.name, asccidoc_type="nt:file")
-            # This is needed to add a new module version, otherwise it won't be handled
-            data[":operation"] = "pant:newModuleVersion"
-            files = {'asciidoc': ('asciidoc', open(path, 'rb'), 'text/x-asciidoc')}
-
-            # Minor question: which is correct, text/asciidoc or text/x-asciidoc?
-            # It is text/x-asciidoc. Here's why:
-            # https://tools.ietf.org/html/rfc2045#section-6.3
-            # Paraphrased: "If it's not an IANA standard, use the 'x-' prefix.
-            # Here's the list of standards; text/asciidoc isn't in it.
-            # https://www.iana.org/assignments/media-types/media-types.xhtml#text
-
-            if not args.dry:
-                r = requests.post(url, headers=HEADERS, data=data, files=files, auth=(args.user, pw))
-                _print_response(r.status_code, r.reason)
-            processed_files.append(path)
-            logger.debug('')
-        elif isResource:
+            if target[0] == '/':
+                _error('Absolute symlink paths are unsupported: ' + str(path) + ' -> ' + target)
+            elif not args.dry:
+                symlinkData = {}
+                symlinkData['jcr:primaryType'] = 'nt:unstructured'
+                symlinkData['sling:resourceType'] = 'pant:symlink'
+                symlinkData['pant:target'] = target
```

Comment:
Perhaps we can use sling’s `sling:redirect` Property here. It might yield benefits in the long term.
See: https://sling.apache.org/documentation/bundles/rendering-content-default-get-servlets.html#redirectservlet for more details.

Reply:
I already played with `sling:redirect` and didn't like it. Here's a rough overview of the scenarios I tried:

Access /realNode/redirectNode via browser: works, redirects you to /realNode/realNode2
Access /realNode/redirectNode/realNode2 via browser: doesn't work
Get /realNode/redirectNode via resource resolver: "works" in that you get the redirectNode itself, but if you want the target node, you have to get it manually
Get /realNode/redirectNode/realNode2 via resource resolver: doesn't work
(In the above examples, realNode2 is a child of the target node)

So given the fact that `sling:redirect` offered almost none of the behavior that actually mattered for us, I felt it was best to just ignore it and write our own solution from scratch rather than to wedge in something that isn't a good fit to start with and might bring some unintended side effects with it.

Additionally, if you have a "link" node and a "target" node that are in the *same directory* and you want to use a `sling:redirect` to point from the 1st to the 2nd, you don't specify "target" as the target, you specify "../target" which differs from the posix standard. So even if we wanted to use `sling:redirect`, we'd have to start modifying the target path, and I'd rather not do that unless there's a clear advantage that makes it worth the effort.